junit-runtime/src/main/scala/scala/scalanative/junit/JUnitTask.scala

```diff
@@ -0,0 +1,199 @@
+package scala.scalanative
+package junit
+
+import sbt.testing._
+
+import scala.annotation.tailrec
+import scala.scalanative.reflect.Reflect
+import scala.util.{Failure, Success, Try}
+
+private[junit] final class JUnitTask(val taskDef: TaskDef,
+                                     runSettings: RunSettings)
+    extends Task {
+
+  def tags(): Array[String] = Array.empty
+
+  override def execute(eventHandler: EventHandler,
+                       loggers: Array[Logger]): Array[Task] = {
+    val reporter = new Reporter(eventHandler, loggers, runSettings, taskDef)
+
+    loadBootstrapper(reporter).foreach { bootstrapper =>
+      executeTests(bootstrapper, reporter)
+    }
+
+    Array.empty
+  }
+
+  private def executeTests(bootstrapper: Bootstrapper,
+                           reporter: Reporter): Unit = {
+    reporter.reportRunStarted()
+
+    var failed  = 0
+    var ignored = 0
+    var total   = 0
+
+    @tailrec
+    def runTests(tests: List[TestMetadata]): Try[Unit] = {
+      val (nextIgnored, other) = tests.span(_.ignored)
+
+      nextIgnored.foreach(t => reporter.reportIgnored(Some(t.name)))
+      ignored += nextIgnored.size
+
+      other match {
+        case t :: ts =>
+          total += 1
+
+          val fc = executeTestMethod(bootstrapper, t, reporter)
+          failed += fc
+          runTests(ts)
+
+        case Nil =>
+          Success(())
+      }
+    }
+
+    val result = runTestLifecycle {
+      Success(())
+    } { _ => catchAll(bootstrapper.beforeClass()) } { _ =>
+      runTests(bootstrapper.tests().toList)
+    } { _ => catchAll(bootstrapper.afterClass()) }
+
+    val (errors, timeInSeconds) = result
+
+    errors match {
+      case e :: Nil if isAssumptionViolation(e) =>
+        reporter.reportIgnored(None)
+        ignored += 1
+
+      case es =>
+        failed += es.size
+        reporter.reportErrors("Test ", None, timeInSeconds, es)
+    }
+
+    reporter.reportRunFinished(failed, ignored, total, timeInSeconds)
+  }
+
+  private[this] def executeTestMethod(bootstrapper: Bootstrapper,
+                                      test: TestMetadata,
+                                      reporter: Reporter): Int = {
+    reporter.reportTestStarted(test.name)
+
+    val result = runTestLifecycle {
+      catchAll(bootstrapper.newInstance())
+    } { instance => catchAll(bootstrapper.before(instance)) } { instance =>
+      handleExpected(test.annotation.expected) {
+        catchAll(bootstrapper.invokeTest(instance, test.name)) match {
+          case Success(f) => f.recover { case t => Failure(t) }
```

Comment:
We can escape the unsupported `Future`s here with:
```suggestion
          case Success(f) => f.value.get.flatten
```
so that the rest of `JUnitTask` can stay synchronous.

Since the JUnit plugin will only create `Future.successful` for now, we know that `.value` will always return a non-empty value.